src/WebJobs.Extensions.DurableTask/Options/DurableTaskOptions.cs

```diff
@@ -163,50 +172,40 @@ public void SetDefaultHubName(string hubName)
-            sb.Append(nameof(this.ExtendedSessionsEnabled)).Append(": ").Append(this.ExtendedSessionsEnabled).Append(", ");
-            sb.Append(nameof(this.UseGracefulShutdown)).Append(": ").Append(this.UseGracefulShutdown).Append(", ");
-            if (this.ExtendedSessionsEnabled)
-            {
-                sb.Append(nameof(this.ExtendedSessionIdleTimeoutInSeconds)).Append(": ").Append(this.ExtendedSessionIdleTimeoutInSeconds).Append(", ");
-            }
-
-            if (this.NotificationUrl != null)
+            // Format the options data as JSON in a way that is friendly for technical humans to read.
+            JObject configurationJson = JObject.FromObject(
+                this,
+                new JsonSerializer
+                {
+                    Converters = { new StringEnumConverter() },
+                    ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                });
+
+            // Don't trace the notification URL query string since it may contain secrets.
+            // This is the only property which we expect to contain secrets. Everything else should be *names*
+            // of secrets that are resolved later from environment variables, etc.
+            if (configurationJson.TryGetValue(nameof(this.NotificationUrl), StringComparison.OrdinalIgnoreCase, out JToken notificationUrlJson) &&
+                notificationUrlJson.Type == JTokenType.String &&
+                Uri.TryCreate((string)notificationUrlJson, UriKind.Absolute, out Uri notificationUri))
             {
-                // Don't trace the query string, since that contains secrets
-                string url = this.NotificationUrl.GetLeftPart(UriPartial.Path);
-                sb.Append(nameof(this.NotificationUrl)).Append(": ").Append(url).Append(", ");
+                string sanitizedUrl = notificationUri.GetLeftPart(UriPartial.Path);
+                configurationJson[nameof(this.NotificationUrl)] = sanitizedUrl;
             }
 
-            sb.Append(nameof(this.Notifications)).Append(": { ");
-            this.Notifications.AddToDebugString(sb);
-            sb.Append(" }, ");
-
-            sb.Append(nameof(this.Tracing)).Append(": { ");
-            this.Tracing.AddToDebugString(sb);
-            sb.Append(" }");
-
-            return sb.ToString();
-        }
-
-        private void AppendStorageProviderValuesToDebugString(StringBuilder sb)
-        {
-            sb.Append(nameof(this.StorageProvider)).Append(": { ");
-            foreach (var value in this.StorageProvider)
+            // At this stage the task hub name is expected to have been resolved. However, we want to know
+            // what the original value was in addition to the resolved value, so we're updating the JSON
+            // blob property to use the original, unresolved value.
+            if (configurationJson.TryGetValue(nameof(this.HubName), StringComparison.OrdinalIgnoreCase, out _))
```

Comment:
Wouldn't `this.HubName`, and hence its value in the configuration JSON, always have a value? Or is this call meant to force using the default value?

Reply:
I'm not sure I understand your question. This code does not reference the value of either `this.HubName` or the default value (`this.defaultHubName`). The intent is to log whatever un-resolved value was specified in host.json (including the name of the app settings from which the task hub is derived, if applicable).

Reply:
I guess my point is that `configurationJson` will always have a value for `HubName`, as it is just parsing `this` which will always give you a value for `DurableTaskOptions.HubName`. Because of that, I don't really see the value of the `if` statment since it will always be true. You can just always perform the assignment of `HubName` to `this.originalHubName`.

Reply:
Oh, you're asking why am I am using `TryGetValue` for `HubName` since that property should always be defined. It's just out of habit in writing defensive code. It didn't seem harmful or misleading to keep it.